libvast/vast/synopsis.hpp

```diff
@@ -51,7 +53,8 @@ class synopsis : public caf::ref_counted {
   /// @param op The operator of the predicate.
   /// @param rhs The RHS of the predicate.
   /// @returns The evaluation result of `*this op rhs`.
-  virtual bool lookup(relational_operator op, data_view rhs) const = 0;
+  virtual std::optional<bool> lookup(relational_operator op,
+                                     data_view rhs) const = 0;
```

Comment:
This is the first time I witness `std::optional` in the codebase. So far we use `caf::optional`. Let's stick to one implementation and switch everything to `std::optional` when the time is ripe.